src/avatar_action.cpp

```diff
@@ -979,8 +979,17 @@ void avatar_action::plthrow( avatar &you, item_location loc,
         add_msg( _( "Never mind." ) );
         return;
     }
-
-    item &thrown = *loc;
+    // make a copy and get the original.
+    // the copy is thrown and has its and the originals charges set appropiately
+    // or deleted from inventory if its charges(1) or not stackable.
+    item *orig = loc.get_item();
+    item thrown;
+    if( orig ) {
```

Comment:
i feel like it would be _technically_ better to do `if( !loc ) {return;}` then assign `orig` and `thrown`.

Reply:
if( !loc ) is already checked above.

Reply:
Ah I see, that makes checking get_item() redundant as loc has alreayd been checked, ive removed that now.